charts/argo-controller/templates/service.yaml

```diff
@@ -0,0 +1,27 @@
+---
+{{- if .Values.metrics.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Release.Name }}-metrics-service
+spec:
+  selector:
+    app.kubernetes.io/name: {{ .Release.Name }}
+  ports:
+    - port: {{ .Values.metrics.port }}
+{{- end }}
+---
+{{- if .Values.metrics.enabled }}
```

Comment:
Nitpick: There's no need for the double conditional, a single `if` can output multiple objects.